src/main/kotlin/org/rust/cargo/toolchain/RustToolchain.kt

```diff
@@ -14,8 +14,8 @@ import java.io.File
 
 data class RustToolchain(val location: String) {
 
-    fun looksLikeValidToolchain(): Boolean =
-        File(pathToExecutable(CARGO)).canExecute()
+    fun looksLikeValidToolchain(exec: String = CARGO): Boolean =
```

Comment:
Let's not add this `exec` parameter and instead create `hasExecuteable(exec: String)` *private* helper which can be used by both `looksLikeValidToolchain` and `queryRustupVersion`